core/src/apps/solana/transaction/instruction.py

```diff
@@ -51,8 +51,15 @@ def parse_instruction_data(
         for property_template in property_templates:
             is_included = True
             if property_template.is_optional:
+                if property_template.is_nullable:
+                    raise DataError("Parameter is nullable and optional")
                 is_included = True if reader.get() == 1 else False
 
+            if property_template.is_nullable:
+                is_included = True if reader.get() == 1 else False
+                if not is_included:
```

Comment:
I think a comment here would be helpful. Something along the lines of `A default (null) value is included for nullable properties (e.g. new_authority in Set Authority instruction) even the is_included flag is not set. We can safely discard this value.`.